indico/modules/rb_new/controllers/backend/bookings.py

```diff
@@ -240,6 +241,10 @@ def _process(self, args):
         }
 
         self.booking.modify(data, session.user)
+
+        room = self.booking.room
+        if room.reservations_need_confirmation and self.booking.is_accepted:
```

Comment:
this should also check for `not room.can_be_booked(session.user, ignore_admin=True)` - if the user can create regular bookings for the room anyway, there's no need to reset the booking back to pending (since it probably wasn't pending before either)

Reply:
Okay, I didn't think of this case. Thanks for pointing out!